repos/fdbt-site/src/pages/api/service.ts

```diff
@@ -17,8 +17,27 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
         }
 
         const dataSource = getRequiredSessionAttribute(req, TXC_SOURCE_ATTRIBUTE).source;
+        const fareTypeAttribute = getRequiredSessionAttribute(req, FARE_TYPE_ATTRIBUTE);
+        const isReturn = 'fareType' in fareTypeAttribute && ['period', 'return'].includes(fareTypeAttribute.fareType);
 
         const service = await getServiceByIdAndDataSource(getAndValidateNoc(req, res), serviceId, dataSource);
+        const directions = Array.from(
+            service.journeyPatterns.reduce((set, pattern) => {
+                set.add(pattern.direction);
+                return set;
+            }, new Set<string>()),
+        );
+
+        if (directions.length === 1 && isReturn) {
+            const errors: ErrorInfo[] = [
+                {
+                    id: 'service',
```

Comment:
Do we want to pass back the selected service to use as the default selection? The error message mentions a service, but it wont be shown on the screen.

Reply:
this is done